src/hope_smart_import/readers.py

```diff
@@ -16,54 +16,79 @@ def identity(x: Any) -> Any:
     return x
 
 
-def _read_worksheet(sheet: "Worksheet", start_at: int, headers: bool,
+def _read_worksheet(sheet: "Worksheet", start_at_row: int, has_header: bool,
                     value_mapper: Callable[[Any], Any] = identity) -> Iterable[dict[str, Any]]:
     rows = sheet.rows
 
     field_names = None
     generate_field_names = False
-    if headers:
+    if has_header:
         field_names = [str(cell.value) for cell in next(rows)]
     else:
         generate_field_names = True
 
-    rows = islice(rows, start_at, None)
+    rows = islice(rows, start_at_row, None)
     for row in rows:
         if generate_field_names:
             field_names = [f"column{d + 1}" for d in range(len(row))]
 
         yield dict(zip(field_names, (value_mapper(cell.value) for cell in row)))
 
 
-def open_xls(filepath: str, *, sheet_index: int = 0, start_at: int = 0, headers: bool = True,
-             value_mapper: "ValueMapper" = identity) -> "SheetResult":
+@overload
+def open_xls(filepath: str, *, sheet_index: int = ..., start_at_row: int = ..., has_header: bool = ...,
```

Comment:
do we need @overload for a simple int | str ?

Reply:
@saxix

Reply:
what about:

```
def  open_xls(filepath: str, *, sheet_index_or_name: int|str = 0, start_at: int = 0, headers: bool = True,
             value_mapper: "ValueMapper" = identity) -> "SheetResult":

      if isinstance(sheet_index_or_name, str):
           .....
      elif isinstance(sheet_index_or_name, int):
           .....
	  else:
          raise ValueError(....

```